### ISSUE TITLE
`AttributeBuilder` has no `char_encoding` setter, the attribute name encoding is always UTF-8

### ISSUE THREAD

**Issue by CramBL** [metno/hdf5-rust#213]
## Current behaviour

`AttributeBuilderInner::create` (`src/hl/attribute.rs`) creates an attribute creation property list and calls `H5Pset_char_encoding(acpl, H5T_CSET_UTF8)` on it before `H5Acreate2`. The list is not reachable from outside the builder. Groups, datasets and links expose the choice through `GroupBuilder::char_encoding`, `DatasetBuilder::char_encoding` and `LinkCreateBuilder::char_encoding` with the `CharEncoding` enum. The C default for an attribute creation property list is `H5T_CSET_ASCII` (see: https://support.hdfgroup.org/documentation/hdf5/latest/group___a_c_p_l.html).

libhdf5 selects the attribute message version from the encoding. An ASCII name under the earliest library bounds gives a version 1 message. A UTF-8 name gives a version 3 message regardless of the bounds (see: https://support.hdfgroup.org/documentation/hdf5/latest/_f_m_t3.html#subsubsec_fmt3_dataobject_hdr_msg_attribute).

### HDF5 1.6 and older readers

Version 3 attribute messages were introduced in HDF5 1.8. A file written with `libver_earliest` for a 1.6 reader still gets version 3 attribute messages, and that reader cannot decode any attribute in it. No error is raised on the writing side.

### HDF5 1.8 and later readers

Every reader from 1.8 on decodes version 3 messages, so there is no interoperability defect. Two smaller effects remain:

- The stored encoding is a property consumers can read through `H5Aget_create_plist` and `H5Pget_char_encoding`, and a name that is ASCII is declared UTF-8. The declaration is valid, and tools that branch on it treat the name as UTF-8, which for ASCII bytes changes nothing.
- A test suite for an HDF5 reader cannot obtain a version 1 or version 2 attribute message from this crate. It needs `H5Pcreate`, `H5Pset_char_encoding` and `H5Acreate2` through FFI for that fixture (**This is the motivating case for this request**)

## Proposed API

```rust
impl AttributeBuilder {
    /// Sets the character encoding of the attribute name.
    ///
    /// Defaults to [`CharEncoding::Utf8`]. With [`CharEncoding::Ascii`] and the
    /// earliest library bounds, libhdf5 writes a version 1 attribute message.
    pub fn char_encoding(self, encoding: CharEncoding) -> Self;
}
```

Carried through `AttributeBuilderEmpty`, `AttributeBuilderEmptyShape` and `AttributeBuilderData` the way `packed` is, with `create` passing `H5T_CSET_ASCII` or `H5T_CSET_UTF8` to `H5Pset_char_encoding`. A test beside `test_create` can create one attribute per encoding under `libver_earliest` and read the encoding back through `H5Aget_create_plist` and `H5Pget_char_encoding`.

## Reproducer for the current gap

```rust
use hdf5_metno as hdf5;

#[test]
fn an_ascii_attribute_name_cannot_be_requested() -> hdf5::Result<()> {
    let dir = tempfile::tempdir().unwrap();
    let path = dir.path().join("earliest.h5");
    let file = hdf5::File::with_options().with_fapl(|p| p.libver_earliest()).create(&path)?;
    file.new_attr::<i32>().shape(()).create("label")?.write_scalar(&1)?;
    file.close()?;

    // The attribute message starts with its version byte. It is 3 although the
    // file was requested in the earliest format, and no builder method changes it.
    let bytes = std::fs::read(&path).unwrap();
    let version_3_message = bytes.windows(14).any(|w| w[0] == 3 && &w[9..14] == b"label");
    assert!(version_3_message);
    Ok(())
}
```
